numpy/core/src/multiarray/abstractdtypes.c

```diff
@@ -126,43 +140,147 @@ initialize_and_map_pytypes_to_dtypes()
 }
 
 
+/*
+ * The following functions define the "common DType" for the abstract dtypes.
+ *
+ * Note that the logic with respect to the "higher" dtypes such as floats
+ * could likely be more logically defined for them, but since NumPy dtypes
+ * largely "know" each other, that is not necessary.
+ */
+static PyArray_DTypeMeta *
+int_common_dtype(PyArray_DTypeMeta *NPY_UNUSED(cls), PyArray_DTypeMeta *other)
+{
+    if (other->legacy && other->type_num < NPY_NTYPES) {
+        if (other->type_num == NPY_BOOL) {
+            /* Use the default integer for bools: */
+            return PyArray_DTypeFromTypeNum(NPY_LONG);
```

Comment:
Is there a NPY_DEFAULT_INT type? Also - is coverage correct that this case is not tested?

Reply:
Added the coverage.  No, we don't have such a macro.  There are two places that three places currently where we use a default I think:

* Here
* In this file, but in the `discover_descriptor_from_pyint`
* In the ufunc code when specialising `np.add.reduce` and `np.multiply.reduce` to always return integers.